Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -125,16 +125,16 @@ Element* ScrollTimeline::source() const
     case Scroller::Nearest: {
         if (CheckedPtr subjectRenderer = source->renderer()) {
             if (CheckedPtr nearestScrollableContainer = subjectRenderer->enclosingScrollableContainer()) {
-                if (RefPtr nearestSource = nearestScrollableContainer->element()) {
-                    Ref document = nearestSource->document();
+                if (auto* nearestSource = nearestScrollableContainer->element()) {
+                    Ref document = Ref { *nearestSource }->document();
                     RefPtr documentElement = document->documentElement();
```

Comment:
I wonder if this needs to be RefPtr if we only use it for comparisons. Though sometimes it does complain about unprotected variables...

Reply:
Pointer comparison is incorrect after free since addresses can be reused. It won't do nonsense, but we might as well prohibit it.

But also: This is C++! Pointer comparison after free is defined to be undefined behavior, so it actually _can_ do nonsense!

Also, our analysis is syntax-only, not data flow, so we do not have a check that all uses are trivial. (Such a check is pretty challenging to implement correctly in C++ since pointers can copy and alias each other.)

So, yeah, Ref.